### ISSUE TITLE
Repairing damage leads to reinforced display glitch

### ISSUE THREAD

**Issue by woutershep** [CleverRaven/Cataclysm-DDA#24446]
Game version: af2663849805534a8a2e941ecdcb149c176e0f4f
Operating system: Linux
Tiles or curses: ncursesw

### Expected behavior

Repairing items should fix one level of damage and the name should be proper.

### Actual behavior

One level of damage is fixed but the item's display name becomes 'reinforced'

### Steps to reproduce the behavior

1. Get a lightly damaged piece of clothing or a weapon.
2. Apply the sewing kit or soldering iron.
3. Press 4 to repeat until fully repaired to a normal state.
4. Press escape.
5. Your item now displays as 'reinforced' despite not being reinforced.

Saving and reloading returns the name to its proper damage level.

Configuration that might be relevant: Item health bars: False; Item symbols: False.

**Comment by Nerezza** [comment]
Sometimes when you repair items, you'll do so good that your repairs bring the item to better than new condition. There's no feedback to indicate this critical success while repairing so it can be confusing as to how you seemingly reinforced something instead of repairing it

**Comment by woutershep** [comment]
> Sometimes when you repair items, you'll do so good that your repairs bring the item to better than new condition. There's no feedback to indicate this critical success while repairing so it can be confusing as to how you seemingly reinforced something instead of repairing it

Interesting point but I'm pretty confident this is not the case here, the dialog still shows 'Reinforce' instead of 'Nothing'/'Practice' and reloading the game drops the 'reinforced' prefix.

**Comment by tinukedaya** [comment]
There is something more to this... I can't repair some things to "normal" fixed state. Even thought the message says I repaired my guns completely they still show as |\  and when I repair them again I get the same message and they go ++ but light green, not dark for reinforced items. 

![image](https://user-images.githubusercontent.com/19891532/43197904-00feab96-900d-11e8-9ff6-1c2dad8a6d80.png)

Same happened to my books that I repaired earlier.
